analyses/ewings-cell-type/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,493 @@
+
+# plot mean gene expression 
+ggplot(marker_sum_exp, aes(x = UMAP1, y = UMAP2, color = sum_exp)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  scale_color_viridis_c() +
+  theme_bw()
+```
+Similar to the individual plots, it looks like there is one group of cells on the bottom right that has the highest marker gene expression. 
+We would anticipate that these are most likely to be the tumor cells. 
+
+Now let's classify any cell that has a sum of marker genes > 0 (after z-transformation) as tumor cells. 
+
+```{r}
+# classify tumor cells based on presence of any marker genes 
+marker_sum_exp <- marker_sum_exp |> 
+  dplyr::mutate(sum_classification = dplyr::if_else(sum_exp > 0, "Tumor", "Normal"))
+
+ggplot(marker_sum_exp, aes(x = UMAP1, y = UMAP2, color = sum_classification)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  theme_bw()
+```
+
+This gives us a rough idea of cells that may be classified as tumor cells. 
+Let's look at the expression of each marker gene in tumor vs. normal cells and see if we see distinct separation between all of them or only a subset. 
+
+```{r}
+# add cell types from sum classification to plotting df 
+celltype_df <- umap_df |> 
+  dplyr::left_join(marker_sum_exp)
+
+# make plot with gene on y-axis, expression on x-axis and then color density by tumor/normal 
+# do all marker genes so differences between tumor/normal or only a subset? 
+ggplot(celltype_df, aes(x=transformed_gene_expression, color = sum_classification)) +
+  geom_density() +
+  facet_wrap(vars(gene_symbol)) +
+  theme_bw()
+```
+
+It looks like ADGRG2, HES1, NROB1, and STEAP1 don't show a lot of difference between cells that might be tumor vs. normal cells and are generally lowly expressed.
+CD99 appears to be expressed in both normal and tumor at similar levels. 
+All other genes show differences between the two, although at varying degrees. 
+Again, we should make sure this is consistent across multiple patients before removing these genes as marker genes. 
+
+### Marker genes using `findMarkers`
+
+Below, we will use `scran::findMarkers()` to find any markers between tumor and normal cells. 
+Since we used marker genes to classify tumor cells, I would expect that at least some of the top markers would be from our list of tumor markers. 
+
+```{r}
+# add in tumor/normal classification to colData of SCE 
+celltypes_only_df <- marker_sum_exp |> 
+  dplyr::select(barcodes, sum_classification)
+
+coldata_df <- as.data.frame(colData(sce)) |> 
+  dplyr::left_join(celltypes_only_df)
+
+colData(sce) <- DataFrame(coldata_df, row.names = rownames(coldata_df))
+
+# find markers between tumor and normal 
+markers <- scran::findMarkers(sce,
+                              groups = sce$sum_classification,
+                              pval.type = "all") 
+```
+
+```{r}
+# pull out top markers for tumor cells 
+tumor_markers <- markers$Tumor |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("ensembl_gene_id") |> 
+  dplyr::left_join(marker_genes_df) |>
```

Comment:
No, I'm doing this so that I can print them out with the gene symbols rather than the IDs. Not all of us have them memorized! I'll add a comment here to state that.